patches/tModLoader/Terraria/ModLoader/ModAccessorySlot.cs

```diff
@@ -0,0 +1,116 @@
+		public virtual string VanityBackgroundTexture => null;
+		public virtual string FunctionalBackgroundTexture => null;
+
+		// Properties to change the Default Style Image Texture
+		public virtual string DyeTexture => null;
+		public virtual string VanityTexture => null;
+		public virtual string FunctionalTexture => null;
+
+		// Properties to control which parts of accessory slot draw
+		public virtual bool SkipUIDrawWileTrue => false;
+		public virtual bool DrawFunctionalSlot => true;
+		public virtual bool DrawVanitySlot => true;
+		public virtual bool DrawDyeSlot => true;
+
+		// Get/Set Properties for fetching slot information
+		public Item FunctionalItem {
+			get => ModSlotPlayer.exAccessorySlot[Type];
+			set => ModSlotPlayer.exAccessorySlot[Type] = value;
+		}
+
+		public Item VanityItem {
+			get => ModSlotPlayer.exAccessorySlot[Type + ModSlotPlayer.SlotCount()];
+			set => ModSlotPlayer.exAccessorySlot[Type + ModSlotPlayer.SlotCount()] = value;
+		}
+
+		public Item DyeItem {
+			get => ModSlotPlayer.exDyesAccessory[Type];
+			set => ModSlotPlayer.exDyesAccessory[Type] = value;
+		}
+
+		public bool ShowVisuals {
+			get => ModSlotPlayer.exHideAccessory[Type];
+			set => ModSlotPlayer.exHideAccessory[Type] = value;
+		}
+
+		protected sealed override void Register() => Type = LoaderManager.Get<AccessorySlotLoader>().Register(this);
+
+		public bool IsEmpty() => FunctionalItem.IsAir && VanityItem.IsAir && DyeItem.IsAir;
+
+		/// <summary>
+		/// This function allows for advanced custom drawing for the accessory slot. Called for Dyes, Vanity, and Functionals.
+		/// By default runs ItemSlot.Draw()
+		/// Receives data:
+		/// <para><paramref name="inv"/> :: the array containing all accessory slots, yours is inv[slot] </para>
+		/// <para><paramref name="slot"/> :: which is the index for inventory that you were assigned </para>
+		/// <para><paramref name="position"/> :: is the position of where the ItemSlot will be drawn </para>
+		/// <para><paramref name="context"/> :: 12 => dye; 11 => vanity; 10 => functional </para>
+		/// </summary>
+		public virtual void DrawModded(Item[] inv, int context, int slot, Vector2 position) =>
+			ItemSlot.Draw(Main.spriteBatch, inv, context, slot, position);
+
+		/// <summary>
+		/// Override to set conditions on when the slot is valid for stat/vanity calculations and player usage.
+		/// Example: the demonHeart is consumed and in Expert mode in Vanilla.
+		/// </summary>
+		public virtual bool IsEnabled() => true;
+
+		/// <summary>
+		/// Override to replace the vanilla effect behavior of the slot with your own.
+		/// By default calls:
+		/// Player.VanillaUpdateEquips(FunctionalItem), Player.ApplyEquipFunctional(FunctionalItem, ShowVisuals), Player.ApplyEquipVanity(VanityItem)
+		/// </summary>
+		public virtual void ApplyEquipEffects() {
+			Player.VanillaUpdateEquip(FunctionalItem);
+			Player.ApplyEquipFunctional(FunctionalItem, ShowVisuals);
```

Comment:
`ApplyEquipFunctional` takes a second parameter `hideVisual`, so it should be `!ShowVisuals`

Reply:
Good catch! Property itself is mis-named, should be HideVisuals as true is invisible